torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -4924,6 +4977,34 @@ def forward(self, x, is_rref=False):
 
 class TensorPipeAgentCudaRpcTest(RpcAgentTestFixture):
 
+    @skip_if_lt_x_gpu(2)
+    def test_cuda_future_extraction_sparse_tensor(self):
+        try:
+            opts = rpc.TensorPipeRpcBackendOptions(init_method=self.init_method)
+            if self.rank == 1:
+                opts.set_device_map("server", {self.rank: 0})
+                rpc.init_rpc(
+                    "master",
+                    rank=self.rank,
+                    world_size=2,
+                    rpc_backend_options=opts
+                )
+                rref = rpc.remote("server", Server)
+                tensor = build_sparse_tensor()
+                tensor = sparse_tensor_to_rpc_format(tensor)
+                fut = rref.rpc_async().identity(tensor).then(sparse_callback)
+                fut.wait()
+            elif self.rank == 0:
+                rpc.init_rpc(
+                    "server",
+                    rank=self.rank,
+                    world_size=2,
+                    rpc_backend_options=opts
+                )
+        finally:
+            if self.rank < 2:
+                rpc.shutdown()
```

Comment:
This test is doing a lot more than strictly necessary: there's no need to spin up the RPC agent and perform network requests just to test the DataPtr extraction logic. If you look at the other tests I pointed at they are much simpler: they create a future and inspect its result to ensure the correct DataPtrs were extracted and used.

Reply:
Okay yeah think I had a build problem. I will update it to use the cuda stream.

Reply:
I updated `_test_cuda_future_extraction` to add to a tensor on one stream because `fill_` is not implemented for sparse tensors and to have the same test for dense and sparse tensors.

```
NotImplementedError: Could not run 'aten::fill_.Scalar' with arguments from the 'SparseCUDA' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build). If you are a Facebook employee using PyTorch on mobile, please visit https://fburl.com/ptmfixes for possible resolutions. 'aten::fill_.Scalar' is only available for these backends: [CPU, CUDA, Meta, QuantizedCPU, QuantizedCUDA, BackendSelect, Named, Conjugate, ADInplaceOrView, AutogradOther, AutogradCPU, AutogradCUDA, AutogradXLA, UNKNOWN_TENSOR_TYPE_ID, AutogradMLC, AutogradHPU, AutogradNestedTensor, AutogradPrivateUse1, AutogradPrivateUse2, AutogradPrivateUse3, Tracer, UNKNOWN_TENSOR_TYPE_ID, Autocast, Batched, VmapMode].
```